new_session_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tidb_test
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"sync"
+
+	"github.com/ngaut/log"
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/store/tikv"
+	"github.com/pingcap/tidb/store/tikv/mock-tikv"
+	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/util/testleak"
+)
+
+var _ = Suite(&testSessionSuite{})
+
+type testSessionSuite struct {
+	cluster   *mocktikv.Cluster
+	mvccStore *mocktikv.MvccStore
+	store     kv.Storage
+}
+
+func (s *testSessionSuite) SetUpSuite(c *C) {
+	s.cluster = mocktikv.NewCluster()
+	mocktikv.BootstrapWithSingleStore(s.cluster)
+	s.mvccStore = mocktikv.NewMvccStore()
+	store, err := tikv.NewMockTikvStore(
+		tikv.WithCluster(s.cluster),
+		tikv.WithMVCCStore(s.mvccStore),
+	)
+	c.Assert(err, IsNil)
+	s.store = store
+	tidb.SetSchemaLease(0)
+	tidb.SetStatsLease(0)
+	_, err = tidb.BootstrapSession(s.store)
+	c.Assert(err, IsNil)
+	logLevel := os.Getenv("log_level")
+	log.SetLevelByString(logLevel)
+}
+
+func (s *testSessionSuite) TestErrorRollback(c *C) {
+	defer testleak.AfterTest(c)()
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+
+	tk.MustExec("drop table if exists t_rollback")
+	tk.MustExec("create table t_rollback (c1 int, c2 int, primary key(c1))")
+	tk.MustExec("insert into t_rollback values (0, 0)")
+
+	var wg sync.WaitGroup
+	cnt := 4
+	wg.Add(cnt)
+	num := 100
+
+	// retry forever
+	tidb.SetCommitRetryLimit(math.MaxInt64)
```

Comment:
Why not just sets the `unlimitedRetryCount ` as the old test code?

Reply:
This test is in `tidb_test` package, for using mock-tikv and testkit.